timeforplanb123/nornir_cli
# v1.0.0 Release

- update python packages:
  - Click to 8.0.1
  - Nornir to 3.1.1
  - Nornir plugins to latests versions
- update Runbook collections:
  - update custom decorator. Now ctx object has become smarter. Now it is dataclass object that supports the creation of any attributes, their modification and passing between you custom Nornir runbooks:
    - ctx.nornir - default object with Nornir object. It can be Nornir object only
    - ctx.result - default object with Result object. Use it with new print_result, write_result, write_results commands
- update show_inventory command:
  - fix bug with incorrect defaults inventory output with --count option
  - add all parameter for --inventory option
  - fix confirmation for hosts/groups list and hosts/groups/defaults/all inventory:
    - confirmation messages contain the requested parameters
    - confirmation messages are not displayed if the number of items is 0 or 1
- add new print_result, write_result, write_results, write_file commands:
  - work with built-in Nornir plugins and custom commands (Runbook collections)
  - you can customize output and write it to file/files
  - commands includes many useful options:
    - --attributes (Result class attributes you want to print or write)
    - --count (limit results)
    - --no_errors (don't write errors to file/files)
    - --diff_to_file or --print_diff (write diff to file or output diff)
    - etc.
- add new change_credentials command:
  - command changes credentials for defaults, hosts/host list, groups/groups list at any time
  - you can use environment variables
- add new Nornir plugins:
  - nornir_routeros
  - nornir_paramiko
  - nornir_http
  - nornir_pyxl
  - nornir_netconf
- apply default value for click.confirm. To press enter is enough
- update functions in nornir_cli.common_plugins:
  - _json_loads. Add parameter type check
  - _doc_generator. Add new rule for docstrings
  - custom. Now it returns new dataclass object
  - multiple_progress_bar is now available from nornir_cli.common.plugins
  - _info is print_stat now
- update documentation
  - https://timeforplanb123.github.io/nornir_cli/
